pkg/collector/corechecks/servicediscovery/envs/envs.go

```diff
@@ -0,0 +1,112 @@
+// Unless explicitly stated otherwise all files in this repository are licensed
+// under the Apache License Version 2.0.
+// This product includes software developed at Datadog (https://www.datadoghq.com/).
+// Copyright 2024-present Datadog, Inc.
+
+// Package envs provides target environment variables of interest.
+package envs
+
+import (
+	"fmt"
+
+	"github.com/DataDog/datadog-agent/pkg/collector/corechecks/servicediscovery/language"
+)
+
+// envKind separates environment variables by type, it helps to know when all target variables
+// have been found and there is no need to search for variables further.
+type envKind uint8
+
+const (
+	empty      envKind = 0
+	pwdEnv             = 1 << (iota - 1) // system settings env variable detected
+	appEnv                               // application system settings env variable detected
+	ddInjected                           // agent injection env variable detected
+	ddService                            // agent service env variable detected
+)
+
+type envLang struct {
+	kind envKind
+	lang language.Language
+}
+
+// targets is a collection of environment variables of interest.
+var targets = map[string]envLang{
```

Comment:
This should change to `map[string]struct{}` when the envLang stuff is removed.

Reply:
this struct has been changed in last commit